app/Controller/CoPetitionsController.php

```diff
@@ -2038,10 +2045,16 @@ function isAuthorized() {
       $args['contain'] = false;
       
       $pt = $this->CoPetition->find('first', $args);
-      
+
       if(!$pt) {
-        $this->Flash->set(_txt('er.notfound', array(_txt('ct.co_petitions.1', $petitionId))), array('key' => 'error'));
-        $this->redirect('/');
+        if($this->request->is('restful')) {
+          $this->Api->restResultHeader(404, "Not Found");
```

Comment:
may add something like in the line 2055 _txt('er.notfound', array(_txt('ct.co_petitions.1', $petitionId))) ?

Reply:
Done